vms/proposervm/block/build.go

```diff
@@ -18,17 +18,30 @@ func BuildUnsigned(
 	parentID ids.ID,
 	timestamp time.Time,
 	pChainHeight uint64,
+	epoch Epoch,
 	blockBytes []byte,
 ) (SignedBlock, error) {
-	var block SignedBlock = &statelessBlock{
-		StatelessBlock: statelessUnsignedBlock{
+	var (
+		statelessUnsignedBlock = statelessUnsignedBlock{
 			ParentID:     parentID,
 			Timestamp:    timestamp.Unix(),
 			PChainHeight: pChainHeight,
 			Certificate:  nil,
 			Block:        blockBytes,
-		},
-		timestamp: timestamp,
+		}
+		block SignedBlock
+	)
+	if epoch.Number == 0 {
```

Comment:
Do we need to use the number as a special field? I feel like it would be a bit more clear (at least within this PR) to do `if epoch == (Epoch{}) {` rather than `if epoch.Number == 0`.